core/src/apps/cardano/sign_tx.py

```diff
@@ -556,26 +687,32 @@ def _validate_stake_pool_registration_tx_structure(msg: CardanoSignTxInit) -> No
         msg.certificates_count != 1
         or msg.signing_mode != CardanoTxSigningMode.POOL_REGISTRATION_AS_OWNER
         or msg.withdrawals_count != 0
+        or msg.additional_witness_requests_count != 0
+        or msg.has_token_minting
     ):
         raise INVALID_STAKE_POOL_REGISTRATION_TX_STRUCTURE
 
 
 def _validate_output(
-    output: CardanoTxOutput, protocol_magic: int, network_id: int
+    keychain: seed.Keychain,
+    output: CardanoTxOutput,
+    signing_mode: CardanoTxSigningMode,
+    protocol_magic: int,
+    network_id: int,
 ) -> None:
     if output.address_parameters and output.address is not None:
-        raise wire.ProcessError(
-            "Outputs can not contain both address and address_parameters fields!"
-        )
+        raise INVALID_OUTPUT
 
-    if output.address_parameters:
-        validate_address_parameters(output.address_parameters)
+    if address_parameters := output.address_parameters:
+        if signing_mode == CardanoTxSigningMode.MULTISIG_TRANSACTION:
```

Comment:
Is there any harm in allowing them? I checked and there also aren't any tests, either positive or negative, covering change outputs with pool registration so perhaps it's implicitly allowed and should either be explicitly allowed or forbidden. I am not sure which though.

Reply:
I would go with forbidden. Letting Trezor/Ledger do crypto derivations (path to key hash) that the user in not aware of (since the outputs are not shown) is a bad practice because it's a potential attack surface.

Reply:
https://github.com/vacuumlabs/trezor-firmware/pull/66/commits/5b5b12948a9a94557e9405af249d0100171aecb8